rclcpp/test/rclcpp/test_rosout_subscription.cpp

```diff
@@ -178,3 +178,13 @@ TEST_F(TestRosoutSubscription, test_rosoutsubscription_getchild_hierarchy) {
   EXPECT_TRUE(future.get());
   received_msg_promise = {};
 }
+
+TEST_F(TestRosoutSubscription, test_rosoutsubscription_node_rosout_disabled) {
+  rclcpp::NodeOptions options = rclcpp::NodeOptions().enable_rosout(false);
+  auto node = std::make_shared<rclcpp::Node>("my_node", options);
+  auto log_func = [&node] {
+      auto logger = node->get_logger().get_child("child");
+      RCLCPP_INFO(logger, "test");
+    };
+  ASSERT_NO_THROW(log_func());
```

Comment:
I need a lambda to call `RCLCPP_INFO` for `ASSERT_NO_THROW` because there is a compilication issue when I tested below,
```cpp
  ASSERT_NO_THROW(
  {
    auto logger = node->get_logger().get_child("child");
    RCLCPP_INFO(logger, "test");
  });
```

<details><summary> RCLCPP_INFO not worked in ASSERT_NO_THROW </summary>
<p>

```shell
In file included from /home/chenlh/Projects/ROS2/ros2-master/local_build_test/rclcpp/include/rclcpp/logging.hpp:24,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/include/rclcpp/copy_all_parameter_values.hpp:27,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/include/rclcpp/rclcpp.hpp:171,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:21:
/home/chenlh/Projects/ROS2/ros2-master/local_build_test/rclcpp/include/rclcpp/logging.hpp:527:5: error: macro "GTEST_TEST_NO_THROW_" passed 4 arguments, but takes just 2
  527 |     RCUTILS_LOG_INFO_NAMED( \
      |     ^~~~~~~~~~~~~~~~~~~~~~
/home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:189:5: note: in expansion of macro ‘RCLCPP_INFO’
  189 |     RCLCPP_INFO(logger, "test");
      |     ^~~~~~~~~~~
In file included from /home/chenlh/Projects/ROS2/ros2-master/local_install_test/gtest_vendor/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:15:
/home/chenlh/Projects/ROS2/ros2-master/local_install_test/gtest_vendor/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1461: note: macro "GTEST_TEST_NO_THROW_" defined here
 1461 | #define GTEST_TEST_NO_THROW_(statement, fail) \
      | 
In file included from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:15:
/home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp: In member function ‘virtual void TestRosoutSubscription_test_rosoutsubscription_node_rosout_disabled_Test::TestBody()’:
/home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:185:3: error: ‘GTEST_TEST_NO_THROW_’ was not declared in this scope
  185 |   ASSERT_NO_THROW(
      |   ^~~~~~~~~~~~~~~
gmake[2]: *** [test/rclcpp/CMakeFiles/test_rosout_subscription.dir/build.make:76: test/rclcpp/CMakeFiles/test_rosout_subscription.dir/test_rosout_subscription.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4078: test/rclcpp/CMakeFiles/test_rosout_subscription.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
--- stderr: rclcpp
In file included from /home/chenlh/Projects/ROS2/ros2-master/local_build_test/rclcpp/include/rclcpp/logging.hpp:24,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/include/rclcpp/copy_all_parameter_values.hpp:27,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/include/rclcpp/rclcpp.hpp:171,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:21:
/home/chenlh/Projects/ROS2/ros2-master/local_build_test/rclcpp/include/rclcpp/logging.hpp:527:5: error: macro "GTEST_TEST_NO_THROW_" passed 4 arguments, but takes just 2
  527 |     RCUTILS_LOG_INFO_NAMED( \
      |     ^~~~~~~~~~~~~~~~~~~~~~
/home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:189:5: note: in expansion of macro ‘RCLCPP_INFO’
  189 |     RCLCPP_INFO(logger, "test");
      |     ^~~~~~~~~~~
In file included from /home/chenlh/Projects/ROS2/ros2-master/local_install_test/gtest_vendor/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:15:
/home/chenlh/Projects/ROS2/ros2-master/local_install_test/gtest_vendor/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1461: note: macro "GTEST_TEST_NO_THROW_" defined here
 1461 | #define GTEST_TEST_NO_THROW_(statement, fail) \
      | 
In file included from /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:15:
/home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp: In member function ‘virtual void TestRosoutSubscription_test_rosoutsubscription_node_rosout_disabled_Test::TestBody()’:
/home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/test/rclcpp/test_rosout_subscription.cpp:185:3: error: ‘GTEST_TEST_NO_THROW_’ was not declared in this scope
  185 |   ASSERT_NO_THROW(
      |   ^~~~~~~~~~~~~~~
gmake[2]: *** [test/rclcpp/CMakeFiles/test_rosout_subscription.dir/build.make:76: test/rclcpp/CMakeFiles/test_rosout_subscription.dir/test_rosout_subscription.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4078: test/rclcpp/CMakeFiles/test_rosout_subscription.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< rclcpp [5.24s, exited with code 2]

```

</p>
</details>